Public/Src/Cache/ContentStore/Library/Utils/ResultsExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Synchronization;
+
+namespace BuildXL.Cache.ContentStore.Utils
+{
+    internal static class ResultsExtensions
+    {
+        /// <nodoc />
+        public static TResult WithLockAcquisitionDuration<TResult, TLockKey>(this TResult result, in LockSet<TLockKey>.LockHandle handle)
+            where TResult : ResultBase
+            where TLockKey : IEquatable<TLockKey>
+        {
+            if (handle.LockAcquisitionDuration != null)
+            {
+                var message = $", lock acquired by {(long)handle.LockAcquisitionDuration.Value.TotalMilliseconds}ms, ";
+                result.ExtraDiagnosticsMessage = message;

```

Comment:
>ExtraDiagnosticsMessage [](start = 23, length = 23)

Should we have an assert that it is not alredy set and being overwritten?

Reply:
Sounds like a good idea.